pdns/dnsdist-lua.cc

```diff
@@ -254,18 +225,18 @@ static void setupLuaConfig(LuaContext& luaCtx, bool client, bool configCheck)
       setLuaSideEffect();
 
       std::shared_ptr<DownstreamState> ret = std::make_shared<DownstreamState>(ComboAddress());
-      newserver_t vars;
+      boost::optional<newserver_t> vars;
 
       ComboAddress serverAddr;
       std::string serverAddressStr;
       if(auto addrStr = boost::get<string>(&pvars)) {
         serverAddressStr = *addrStr;
         if(qps) {
-          vars["qps"] = std::to_string(*qps);
+          (*vars)["qps"] = std::to_string(*qps);
```

Comment:
Does that work? I would expect that de-referencing an unset `boost::optional` would be a problem.

Reply:
I have to try.